infrastructure/modules/serverless/iam-role.tf

```diff
@@ -474,7 +475,11 @@ resource "null_resource" "delete_zips" {
     lambda_function_arn = "arn:aws:lambda:${var.region}:${var.account_id}:function:${each.key}"
   }
 
+  depends_on = [aws_lambda_function.IamRolesService, aws_lambda_function.InstanceChangeState, aws_lambda_function.IamRolesServiceMapping,
+  aws_lambda_function.IamRolestoGrafana]
+
```

Comment:
@waji46 remove all the null_resources and use terraform_data resource.

Reply:
Replaced null_resources with the terraform_data resource